unittests/misc_tests.cpp

```diff
@@ -826,6 +834,35 @@ BOOST_AUTO_TEST_CASE(transaction_test) { try {
    BOOST_CHECK( packed != fc::raw::pack( static_cast<const transaction&>(pkt8.get_transaction()) ));
    BOOST_CHECK( packed == pkt8.get_packed_transaction() ); // extra maintained
 
+   // Test extra maintained in non-empty context_free_data transaction
+   // in transitions of v0 -> v1 -> v0
+   packed_transaction pkt8_v0_to_v1(pkt8, true); // Build v1 with extra
+   auto pkt8_v0_to_v1_to_v0 = pkt8_v0_to_v1.to_packed_transaction_v0();
+   BOOST_CHECK_EQUAL(pkt.get_transaction().id(), pkt8_v0_to_v1_to_v0->get_transaction().id());
+   BOOST_CHECK( packed != fc::raw::pack( static_cast<const transaction&>(pkt8_v0_to_v1_to_v0->get_transaction()) ));
+   BOOST_CHECK( packed == pkt8_v0_to_v1_to_v0->get_packed_transaction() );
+   verify_packed_transaction_conversion(pkt8, pkt8_v0_to_v1_to_v0);
+
+   // Test empty context_free_data transaction go transitions of v0 -> v1 -> v0
+   signed_transaction empty_cfd_trx;
+   empty_cfd_trx.context_free_actions.push_back({ {}, "eosio"_n, ""_n, bytes() });
+   empty_cfd_trx.context_free_data.push_back(bytes());
+   test.set_transaction_headers(empty_cfd_trx);
+   empty_cfd_trx.sign( test.get_private_key( "eosio"_n, "active" ), test.control->get_chain_id() );
+   packed_transaction_v0 pkt_v0_empty_cfd(empty_cfd_trx);
+   packed_transaction pkt_v0_to_v1_empty_cfd(pkt_v0_empty_cfd, true);
+   auto pkt_v0_to_v1_to_v0_empty_cfd = pkt_v0_to_v1_empty_cfd.to_packed_transaction_v0();
+   verify_packed_transaction_conversion(pkt_v0_empty_cfd, pkt_v0_to_v1_to_v0_empty_cfd);
+
+   // Round trip from packed_transaction to packed_transaction
+   packed_transaction pkt_v0_to_v1(*pkt8_v0_to_v1_to_v0, true);
+   BOOST_CHECK_EQUAL(pkt.get_transaction().id(), pkt_v0_to_v1.get_transaction().id());
+   BOOST_CHECK( packed != fc::raw::pack( static_cast<const transaction&>(pkt_v0_to_v1.get_transaction()) ));
+   BOOST_CHECK( packed == pkt_v0_to_v1.get_packed_transaction() );
+   BOOST_CHECK_EQUAL(pkt_v0_to_v1.get_prunable_size(), pkt_v0_to_v1.get_prunable_size());
+   BOOST_CHECK_EQUAL(pkt_v0_to_v1.get_unprunable_size(), pkt_v0_to_v1.get_unprunable_size());
+   BOOST_REQUIRE_EQUAL(pkt_v0_to_v1.get_context_free_data()->size(), pkt_v0_to_v1.get_context_free_data()->size());
+   BOOST_CHECK(std::equal(pkt_v0_to_v1.get_context_free_data()->begin(), pkt_v0_to_v1.get_context_free_data()->end(), pkt_v0_to_v1.get_context_free_data()->begin()));
```

Comment:
It should be possible to have extra data at the end of `packed_context_free_data` just like you can have extra data at the end of packed_trx. Please add some tests for that as well.

Reply:
Extra data was added to packed_context_free_data in tests. Local variables were renamed to help understand the code.